rad/src/operators/string.rs

```diff
@@ -127,7 +130,27 @@ fn json_to_cbor(value: &json::JsonValue) -> Value {
         }
         json::JsonValue::Short(value) => Value::Text(String::from(value.as_str())),
         json::JsonValue::String(value) => Value::Text(String::from(value.as_str())),
-        json::JsonValue::Number(value) => Value::Float((*value).into()),
+        json::JsonValue::Number(value) => {
+            // This trick allows to tell whether a number has decimals by checking whether the
+            // mantissa contains more significant bits than the exponent can express.
+            let (positive, mantissa, exponent) = value.as_parts();
+            // Factor is 10 ^ |exponent|
+            let factor = u64::pow(
+                10,
+                u32::try_from(exponent.abs())
+                    .expect("Exponent of JsonValue::Number should fit in u32"),
+            );
```

Comment:
`u64::pow` will panic if the result does not fit in a u64, for example when the exponent is 100.

Reply:
Don't think we will ever need 100 as an exponent, because such think doesn't exist in JSON anyway, as it is subject to the limitations of double precision  / binary64 as put forth by IEEE 754 (minimum = `-(2^53 - 1)`, maximum = `2^53 - 1`, exponent_bits = `11`)

Reply:
What do you mean? `1e100` is valid JSON, and `1` followed by 100 zeros is also valid JSON. Add this test:

```rust
        let json = JsonValue::Number(Number::from(1e100));
        let resulting_cbor = json_to_cbor(&json);
        let expected_cbor = serde_cbor::Value::Float(1e100);
        assert_eq!(resulting_cbor, expected_cbor);
```

Reply:
That works now! :wink:

Reply:
Cool, let's add a few tests near `i128::MAX` and `i128::MIN` to make sure it works as expected.